cpp/modmesh/universe/polygon.hpp

```diff
@@ -781,6 +788,617 @@ class TrianglePad
 using TrianglePadFp32 = TrianglePad<float>;
 using TrianglePadFp64 = TrianglePad<double>;
 
+/**
+ * Forward declaration of PolygonPad for use in Polygon3d handle class.
+ */
+template <typename T>
+class PolygonPad;
+
+/**
+ * Polygon3d handle class - lightweight view into a polygon stored in PolygonPad.
+ *
+ * This is a non-owning handle that references a polygon stored in a PolygonPad
+ * container. Polygons are defined by an ordered list of nodes following the
+ * right-hand rule: counter-clockwise for positive area, clockwise for negative.
+ *
+ * The handle uses polygon_id as public API, with internal offset/count for efficient access.
+ *
+ * @tparam T floating-point type
+ */
+template <typename T>
+class Polygon3d
+{
+
+public:
+
+    using point_type = Point3d<T>;
+    using value_type = T;
+    using segment_type = Segment3d<T>;
+    using polygon_pad_type = PolygonPad<T>;
+
+    Polygon3d(std::shared_ptr<polygon_pad_type> pad, size_t polygon_id)
+        : m_pad(std::move(pad))
+        , m_id(polygon_id)
+    {
+    }
+
+    Polygon3d() = default;
```

Comment:
Default constructor of `Polygon3d` does not make sense.  It should be deleted.

Reply:
Deleted